lib/home_page.dart

```diff
@@ -127,6 +140,41 @@ class HomeState extends State<Home> with WidgetsBindingObserver {
       });
     });
 
+    commentBloc = Provider.of<NostrCommentBloc>(context, listen: false);
+
+    widget.marketplaceBloc.nostrSettingsStream.listen((event) {
+      commentBloc.toggleCommentController.add(!event.enableNostr);
+    });
+
+    commentBloc.commentsLoginStream.listen((event) async {
+      NostrSettings settings =
+          await widget.marketplaceBloc.nostrSettingsStream.first;
+
+      if (settings.isLoggedIn == false) {
+        widget.marketplaceBloc.nostrSettingsSettingsSink.add(settings.copyWith(
+          isLoggedIn: true,
+        ));
+      }
+    });
+
+    commentBloc.pubKeyStream.listen((event) async {
+      widget.nostrBloc.actionsSink.add(GetPublicKey());
+
+      final publicKey = await widget.nostrBloc.publicKeyStream.first;
+      commentBloc.getPubKeyResult(publicKey);
+    });
+
+    // Listener for signing the comment
+    commentBloc.signEventStream.listen((event) async {
+      final nostrPrivateKey = widget.nostrBloc.nostrPrivateKey;
+
+      widget.nostrBloc.actionsSink.add(SignEvent(event, nostrPrivateKey));
```

Comment:
It seems that it is more correct to not pass the private key to the SignEvent since the private key is already saved in the bloc. 
All operations should not read (and pass) the private key which should be private field on the bloc and used there when needed.

Reply:
Yes, will make the changes.